### PR TITLE
Lasb 779 - calling authorisation maat-api and web client 

### PR DESCRIPTION
**Pull request by atharmajeed**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/LASB-779)

This PR migrated the stored procedure for a 

## Checklist

Before you ask people to review this PR:

- [ ] Tests should be passing: `./gradlew test`
- [ ] Github should not be reporting conflicts; you should have recently run `git rebase master`.
- [ ] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [ ] You should have looked at the diff against master and ensured that nothing unexpected is included in your changes.
- [ ] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: crime-means-assessment/src/main/java/uk/gov/justice/laa/crime/meansassessment/model/AuthorizationResponse.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package uk.gov.justice.laa.crime.meansassessment.model;
+
+import lombok.*;
+
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+public class AuthorizationResponse {
```
Comment (Mhowell494) [inline crime-means-assessment/src/main/java/uk/gov/justice/laa/crime/meansassessment/model/AuthorizationResponse.java:9]: There is already an AuthorizationResponseDTO class.

Comment (Mhowell494) [inline crime-means-assessment/src/main/java/uk/gov/justice/laa/crime/meansassessment/model/AuthorizationResponse.java:9]: There is already an AuthorizationResponseDTO class.

File: crime-means-assessment/src/main/java/uk/gov/justice/laa/crime/meansassessment/client/AuthorisationMeansAssessmentClient.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import org.springframework.web.reactive.function.client.WebClient;
+import uk.gov.justice.laa.crime.meansassessment.model.AuthorizationResponse;
+
+@Slf4j
+@Component
+public class AuthorisationMeansAssessmentClient {
```
Comment (Mhowell494) [inline crime-means-assessment/src/main/java/uk/gov/justice/laa/crime/meansassessment/client/AuthorisationMeansAssessmentClient.java:12]: I can't see where this class is being used - can it be deleted?
The CreateAssessmentValidator already seems to validate the new work reason.

File: crime-means-assessment/src/main/java/uk/gov/justice/laa/crime/meansassessment/exception/MeansAssessmentValidationException.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package uk.gov.justice.laa.crime.meansassessment.exception;
+
+
+import java.io.Serializable;
+
+public class MeansAssessmentValidationException extends Exception implements Serializable {
```
Comment (Mhowell494) [inline crime-means-assessment/src/main/java/uk/gov/justice/laa/crime/meansassessment/exception/MeansAssessmentValidationException.java:6]: There is already a ValidationException class, do we need another one?

**Review by Mhowell494 — CHANGES_REQUESTED** [review]
Can the changes to the means assessment service be rolled back? I've already refactored the classes involved as part of the work on migrating the full assessment business rules. I could then plug in the new validation code as part of my PR.

File: crime-means-assessment/src/main/java/uk/gov/justice/laa/crime/meansassessment/client/AuthorisationMeansAssessmentClient.java (hunk 2 above)
Comment (atharmajeed) [reply crime-means-assessment/src/main/java/uk/gov/justice/laa/crime/meansassessment/client/AuthorisationMeansAssessmentClient.java:12]: Removed.

File: crime-means-assessment/src/main/java/uk/gov/justice/laa/crime/meansassessment/model/AuthorizationResponse.java (hunk 1 above)
Comment (atharmajeed) [reply crime-means-assessment/src/main/java/uk/gov/justice/laa/crime/meansassessment/model/AuthorizationResponse.java:9]: Okay. removed the duplicated class.

**Pull request merged**